Fix for silently ignored version argument on CentOS/RHEL

pjcreath opened the pull request:
### What does this PR do?
This PR adds a warning when a pre-existing saltstack.repo on CentOS/RHEL causes the version argument to be ignored.

It also allows the -F (forced overwrite) option to override this situation, overwriting the saltstack.repo file, and thus installing the specified version.

### What issues does this PR fix or reference?
Resolves #1209.

pjcreath committed:
bootstrap-salt.sh: fix for silently ignored version argument on CentOS/RHEL

This commit adds a warning when a pre-existing saltstack.repo
on CentOS/RHEL causes the version argument to be ignored.

It also allows the -F (forced overwrite) option to override
this situation, overwriting the saltstack.repo file, and
thus installing the specified version.

Resolves #1209.

vutny reviewed (commented):
This is great!

Also, after creating the repo file, you may want to clean caches (`yum clean metadata`) for some changes to actually take effect.

rallytime commented:
Yes, thanks @pjcreath! Did you see the comment from @vutny above? I think that is a good idea. Once that's implemented, we can get this in. :)

pjcreath committed:
Add 'yum clean metadata' after updating the saltstack.repo file as requested.

According to https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/sec-working_with_yum_cache
this option "eliminates all files that yum uses to determine the remote
availability of packages."

pjcreath commented:
It seemed to work fine without cleaning the caches, at least when the only thing writing the file was bootstrap-salt.sh across varying invocations. I'm guessing it's more generally correct when overwriting the .repo file to clean the caches, so I've added that to the latest commit.

vutny commented:
Yeah, that is only required if you would like to "downgrade" Salt by removing packages and run the bootstrap all over again. If local repo metadata is newer than remote, `yum` will just refuse to install pkgs from the new repo entry.

This is good. Thanks.
